src/main/java/com/epam/ta/reportportal/dao/TestItemRepositoryCustomImpl.java

```diff
@@ -389,6 +391,39 @@ public TestItemTypeEnum getTypeByItemId(Long itemId) {
 		return PageableExecutionUtils.getPage(items, pageable, () -> dsl.fetchCount(QueryBuilder.newBuilder(filter).build()));
 	}
 
+	@Override
+	public List<NestedStep> findAllNestedStepsByIds(Collection<Long> ids) {
+
+		return dsl.select(TEST_ITEM.ITEM_ID,
+				TEST_ITEM.NAME,
+				TEST_ITEM.START_TIME,
+				TEST_ITEM.TYPE,
+				TEST_ITEM_RESULTS.STATUS,
+				TEST_ITEM_RESULTS.END_TIME,
+				TEST_ITEM_RESULTS.DURATION,
+				DSL.field(DSL.exists(dsl.selectOne().from(LOG).where(LOG.ITEM_ID.eq(TEST_ITEM.ITEM_ID)).limit(1))
+						.orExists(dsl.selectOne()
```

Comment:
When you use 'exists' -> you can use simply 'select' without one or limit.
Postgres automatically optimize a query and ends when the first result is found.
prove: https://blog.jooq.org/tag/exists/

Reply:
fixed